### PR TITLE
interface conversion primitives test cases

### PR DESCRIPTION
**Pull request by akrmn**
Follow-up to https://github.com/digital-asset/daml/pull/13391

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala [hunk 1]
```diff
@@ -1116,6 +1202,38 @@ class DecodeV1Spec
         ),
         signatoryInterface -> ESignatoryInterface(ifaceId = scalaIfaceTyConName, body = EUnit),
         observerInterface -> EObserverInterface(ifaceId = scalaIfaceTyConName, body = EUnit),
+        toInterface -> EToInterface(
```
Comment (akrmn) [inline daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:1205]: we were missing a few, thanks for reminding me to check @remyhaemmerle-da

Comment (remyhaemmerle-da) [reply daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:1205]: :bow:

**Review by remyhaemmerle-da — APPROVED** [review]
Nice.

I would love to have also a module in `daml-lf/encoder/src/test/lf/Interface_1.dev_.lf` that contains all the new constructs for interface.

**Review by sofiafaro-da — APPROVED** [review]
Looks great, thank you!

File: daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala [hunk 2]
```diff
@@ -714,6 +714,30 @@ private[daml] class EncodeV1(minor: LV.Minor) {
               .setContractIdExpr(cid)
               .setInterfaceExpr(value)
           )
+        case EInterfaceTemplateTypeRep(iface, value) =>
```
[the remark this answers is not in the document]
Comment (akrmn) [reply daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala:717]: thanks go to @remyhaemmerle-da, I only noticed while writing `daml-lf/encoder/src/test/lf/Interface_1.dev_.lf`, which he requested :)

**Pull request merged**